src/java/org/streamingpool/core/service/streamfactory/FlattenedStreamFactory.java

```diff
@@ -52,6 +52,7 @@
         StreamId<Iterable<T>> source = filteredId.sourceStreamId();
 
         return Optional.of(Flowable.fromPublisher(discoveryService.discover(source))
+                .filter(Objects::nonNull)
                 .flatMap(Flowable::fromIterable)
```

Comment:
Consider checking that none of the elements of the iterable is null and deflecting to the appropriate error stream, otherwise this stream might blow subsequently...